event_app.py

```diff
@@ -1,42 +1,54 @@
 #!/usr/bin/env python
 
+import logging
 from flask import Flask, request, jsonify
 from flask_cors import CORS
 import re
 import json
 from flasgger import Swagger
 
-from ensembl_prodinf import reporting
+from ensembl_prodinf.amqp_publishing import AMQPPublisher
+from ensembl_prodinf.reporting import make_report, ReportFormatter
 from ensembl_prodinf.hive import HiveInstance
 from ensembl_prodinf.event_tasks import process_result
 from ensembl_prodinf.exceptions import HTTPRequestError
 import event_config
 
 
-pool = reporting.get_pool(event_config.report_server)
-
-
-def get_logger():
-    return reporting.get_logger(pool, event_config.report_exchange, 'event_handler', None, {})
+event_formatter = ReportFormatter('event_handler')
+publisher = AMQPPublisher(event_config.report_server,
+                          event_config.report_exchange,
+                          exchange_type=event_config.report_exchange_type,
+                          formatter=event_formatter)
 
 
 app = Flask(__name__, instance_relative_config=True)
 app.config.from_object('event_config')
+app.logger.addHandler(app_logging.default_handler())
 app.config['SWAGGER'] = {
     'title': 'Event App',
     'uiversion': 2
 }
 
 swagger = Swagger(app)
 
-print(app.config)
+app.logger.info(app.config)
+
+
+def log_and_publish(report):
```

Comment:
I am wondering why do you have to redeclare this ? Since the function is in core as well, right ?

Reply:
That function is just a handy shortcut to mimic previous behaviour and hopefully its life will be very short also because it's kind of redundant as `AMQPPublisher` already logs each messages it delivers (in debug mode). Since the function is really simple in my opinion is better to have it there where one can see clearly what the behaviour is, instead of implementing it in a more generic way in a library obfuscating the behaviour, and when we shouldn't maintain it for long. Finally, in this particular case, the event_app is not used AFAIK, so the function is there mostly for exemplification.